File: Selection_Sort.py
def selection_sort(l):
    start = 0
    for start in range(len(l)):
        min_pos = start
        for i in range(start,len(l)):
            if l[min_pos] > l[i]:
                min_pos = i

        (l[start],l[min_pos]) = (l[min_pos],l[start])


arr = [12, 11, 13, 5, 6]
selection_sort(arr)
for i in range(len(arr)):
    print ("% d" % arr[i])

File: selectionSort.c
//program to implement selection sort
#include<stdio.h>

void swap(int *xp, int *yp)
{
	int temp = *xp;
	*xp = *yp;
	*yp = temp;
}

void printArray(int arr[], int SIZE)
{
	int i;
	for(i = 0; i < SIZE; i++)
		printf("%d\n",arr[i]);
}

void selectionSort(int arr[],int n)
{
	int i,j,min_idx;
	for(i = 0; i < n-1; i++)
	{
		//find the minimum element in unsorted array
		min_idx = i;
		for(j = i+1; j < n; j++)
			if(arr[j] < arr[min_idx])
				min_idx = j;
				
		//swap the found minimum element with the first element		
		swap(&arr[min_idx],&arr[i]);
	}
}

int main()
{
	int arr[] = {6,5,4,3,2,1};
	int n = sizeof(arr)/sizeof(arr[0]);
	selectionSort(arr,n);
	printf("Sorted Array : \n");
	printArray(arr,n);
	return 0;
}
